### DISCUSSION TITLE
Dart implementation

### DISCUSSION THREAD

**Discussion by akshayjshah** [connectrpc/.github#8] [Polls]
Would you or your company use a Connect implementation in Dart? If yes, please leave a comment and tell us more about what you'd like (especially if you need integration with a particular framework).

**Comment by debkanchan** [comment]
All of our backend runs on connect. The only problem is our 2 flutter frontends use either gRPC or http. Connect for dart is our #1 requirement rn

**Comment by chimon2000** [comment]
We would love to explore using Connect on the backend. Currently, we are exploring the use of libraries like Dart Frog, which is built on Google's Shelf.  I think integration with Shelf is a definite nice-to-have.

**Comment by Broderick-Westrope** [comment]
I'm quite new to the Connect RPC ecosystem. Having been working with the Go package for a while now I'm seriously loving the work you're doing. The only downside is not having a Connect RPC client in Dart. gRPC will do for now, but I look forward to the future!

Also: unsure if it requires any further integration, but I use Riverpod for state management and would love to see the client be leveraged within its providers. 

Thanks again!

**Comment by galen211** [comment]
@akshayjshah - my company would be willing to contribute resources to / sponsor an official connectrpc dart client.

**Reply by akshayjshah** [reply]
Excellent 🎉  I'd love to chat with you in a bit more detail. If you can join the Buf or CNCF Slack communities (https://buf.build/links/slack or https://communityinviter.com/apps/cloud-native/cncf), ping me and we can set something up!

**Reply by chocolatkey** [reply]
@akshayjshah I am planning to use ConnectRPC with Flutter in an upcoming project. Was there any update on the situation since this chat? I would like to avoid duplicate work if possible

**Comment by akshayjshah** [comment]
Good news: https://github.com/connectrpc/connectrpc.com/pull/239 was just approved, so we'll be shipping a client-only Dart implementation of Connect 🚀

**Comment by nicksnyder** [comment]
Just an update, we have a 0.1.0 Dart implementation here: https://github.com/connectrpc/connect-dart

Please file an issue if you have any problems or feedback!

**Comment by srikrsna-buf** [comment]
We just released a new version (`0.3.0`). It is feature complete and will likely be promoted to a release candidate. 

Please try it out and file an issue if you have any problems or feedback.

**Comment by nicksnyder** [comment]
We haven't gotten any issues filed, but also don't want to mistake lack of feedback for positive feedback. If you are successfully and happily using this package it would be helpful for us to know by replying to this comment or 🚀 reacting to this comment.

Of course if you have any ideas for improvements we could make before a 1.0 release candidate, please [file an issue](https://github.com/connectrpc/connect-dart/issues).
Thank you!

**Reply by torkelrogstad** [reply]
We're using the Dart implementation in a few different desktop applications: https://github.com/LayerTwo-Labs/drivechain-frontends. Happy with it so far!

**Reply by chimon2000** [reply]
When attempting to follow the tutorial, I get the following error with the generated file:

```
The method '$_clearField' isn't defined for the type 'SayResponse'.
Try correcting the name to the name of an existing method, or defining a method named '$_clearField'
```

**Reply by nicksnyder** [reply]
@chimon2000 can you please file a bug report here: https://github.com/connectrpc/connect-dart/issues/new?template=bug_report.md

**Reply by Broderick-Westrope** [reply]
@nicksnyder I finally had a chance to try out the dart client and came across the same error mentioned by @chimon2000. I have created https://github.com/connectrpc/connect-dart/issues/16.

**Comment by chrispine** [comment]
Now that we've released [v1.0.0](https://github.com/connectrpc/connect-dart/releases/tag/v1.0.0), I'm going to close this discussion. If you have any issues or feedback, please let us know there. Thanks!
